workos/sso.py

```diff
@@ -248,80 +217,120 @@ def list_connections(
         Returns:
             dict: Connections response from WorkOS.
         """
-        warn(
-            "The 'list_connections' method is deprecated. Please use 'list_connections_v2' instead.",
-            DeprecationWarning,
-        )
-
-        # This method used to accept `connection_type` as a string, so we try
-        # to convert strings to a `ConnectionType` to support existing callers.
-        #
-        # TODO: Remove support for string values of `ConnectionType` in the next
-        #       major version.
-        if connection_type is not None and isinstance(connection_type, str):
-            try:
-                connection_type = ConnectionType[connection_type]
-
-                warn(
-                    "Passing a string value as the 'connection_type' parameter for 'list_connections' is deprecated and will be removed in the next major version. Please pass a 'ConnectionType' instead.",
-                    DeprecationWarning,
-                )
-            except KeyError:
-                raise ValueError("'connection_type' must be a member of ConnectionType")
-
-        if limit is None:
-            limit = RESPONSE_LIMIT
-            default_limit = True
 
         params = {
-            "connection_type": connection_type.value if connection_type else None,
+            "connection_type": connection_type,
```

Comment:
We are not consistent about when we always add a key (even if it's `None`) or only add the key to the params if it's specified. I'm pretty sure I'm guilty of this as well. We should pick a style and go with it.

Reply:
... and I realized, if we do want to omit missing keys, I wonder if it's easier to come full circle and define our params as `BaseModel`s.

Then we could just....
```python
params = ConnectionParams(connection_type=connection_type)
# when it's time to use the params
Request(params=ConnectionParams.model_dump(exclude_unset=True))
```

Reply:
Oh. In that case I guess we don't need to care.